### ISSUE TITLE
False positive for clippy::inline_modules that only exists in rust-analyzer

### ISSUE THREAD

**Issue by Dryvnt** [rust-lang/rust-analyzer#23224]
Per [clippy docs](https://rust-lang.github.io/rust-clippy/rust-1.98.0/index.html#inline_modules) clippy::inline_modules should not trigger for modules annotated with `#[cfg(test])`. `cargo clippy` correctly follows this, but for some reason rust-analyzer still reports it as a warning.

```text
> cargo clippy
warning: inline module found
 --> src/main.rs:5:1
  |
5 | mod bar {}
  | ^^^^^^^^^^
  |
  = help: move the contents of the module to `src/bar.rs` or `src/bar/mod.rs`, and replace this with `mod bar;`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.98.0/index.html#inline_modules
  = note: requested on the command line with `-W clippy::inline-modules`

warning: `clippy-inline-module-repro` (bin "clippy-inline-module-repro") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.02s
```

<img width="740" height="319" alt="Image" src="https://github.com/user-attachments/assets/dc0b308a-d5a4-4d15-923f-4b264e2a907e" />

**rust-analyzer version**: rust-analyzer version: 0.3.3025-standalone [/home/thomas/.vscode/extensions/rust-lang.rust-analyzer-0.3.3025-linux-x64/server/rust-analyzer] / rust-analyzer version: 0.4.3023-standalone [/home/thomas/.vscode/extensions/rust-lang.rust-analyzer-0.4.3023-linux-x64/server/rust-analyzer]

**rustc version**: rustc 1.98.0 (88d9e12ae 2026-08-18)

**editor or extension**: VSCode, rust-lang.rust-analyzer, version 0.3.3025 / 0.4.3023

**relevant settings**:
.vscode/settings.json
```json
{ "rust-analyzer.check.command": "clippy" }
```
Cargo.toml
```toml
[lints.clippy]
inline_modules = "warn"
```

**code snippet to reproduce**:
```rust
fn main() {
    println!("Hello, world!");
}

mod bar {}
#[cfg(test)]
mod foo {}
```

**Comment by lnicola** [comment]
FWIF, I can't reproduce this:

<img width="411" height="217" alt="Image" src="https://github.com/user-attachments/assets/2128fafd-6f90-4663-8639-b7a1719e3a73" />

**Comment by Dryvnt** [comment]
Huh. I see. It's not impossible that it's something specific to my setup and/or not related to rust-analyzer at all.

I'll try to investigate further. Any potential places it would make sense to look? e.g. how does rust-analyzer call clippy, can I track down the exact binary getting executed?

**Comment by lnicola** [comment]
It should be something like `cargo clippy --all-workspaces`. Do you have any per-user VS Code settings?

**Comment by lnicola** [comment]
Ah, okay, it's a Clippy issue, it didn't happen for me because I'm on `beta`:

```
$ cargo +stable clippy --all-targets
    Checking clippy-test v0.1.0 (~/clippy-test)
warning: inline module found
 --> src/main.rs:5:1
  |
5 | mod bar {}
  | ^^^^^^^^^^
  |
  = help: move the contents of the module to `src/bar.rs` or `src/bar/mod.rs`, and replace this with `mod bar;`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.98.0/index.html#inline_modules
  = note: requested on the command line with `-W clippy::inline-modules`

warning: `clippy-test` (bin "clippy-test") generated 1 warning
warning: inline module found
 --> src/main.rs:7:1
  |
7 | mod foo {}
  | ^^^^^^^^^^
  |
  = help: move the contents of the module to `src/foo.rs` or `src/foo/mod.rs`, and replace this with `mod foo;`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.98.0/index.html#inline_modules

warning: `clippy-test` (bin "clippy-test" test) generated 2 warnings (1 duplicate)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.04s
```

**Comment by Dryvnt** [comment]
Confirmed `--all-targets` triggers it in clippy, this is a clippy issue. Cheers! 👍

```text
> cargo clippy --all-targets
warning: inline module found
 --> src/main.rs:5:1
  |
5 | mod bar {}
  | ^^^^^^^^^^
  |
  = help: move the contents of the module to `src/bar.rs` or `src/bar/mod.rs`, and replace this with `mod bar;`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.98.0/index.html#inline_modules
  = note: requested on the command line with `-W clippy::inline-modules`

warning: inline module found
 --> src/main.rs:7:1
  |
7 | mod foo {}
  | ^^^^^^^^^^
  |
  = help: move the contents of the module to `src/foo.rs` or `src/foo/mod.rs`, and replace this with `mod foo;`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.98.0/index.html#inline_modules

warning: `clippy-inline-module-repro` (bin "clippy-inline-module-repro" test) generated 2 warnings
warning: `clippy-inline-module-repro` (bin "clippy-inline-module-repro") generated 1 warning (1 duplicate)
    Finished [`dev` profile [unoptimized + debuginfo]](https://doc.rust-lang.org/cargo/reference/profiles.html#default-profiles) target(s) in 0.02s
```

**Comment by Dryvnt** [comment]
For anyone else stumbling on this in the future and somehow land here, see https://github.com/rust-lang/rust-clippy/issues/17342 - Fix appears to be coming in 1.99
